scripts/ninja.js

```diff
@@ -585,13 +585,33 @@ function sourceToTarget(y) {
  */
 function ocamlDepForBscAsync(files, dir, depsMap) {
   return new Promise((resolve, reject) => {
+    let mlfiles = [];
+    files.forEach(f => {
+      if (f.endsWith(".res") || f.endsWith(".resi")) {
+        isIntf = f.endsWith(".resi");
+        let mlname = f.slice(0, isIntf ? -5 : -4) + (isIntf ? ".mli" : ".ml");
```

Comment:
nit: `isIntf` seems not defined before. It needs to be defined here.
nit: Using `path.parse` seems more idiomatic.

```suggestion
      const {name, ext} = path.parse(f);
      if (ext === ".res" || ext === ".resi") {
        const isIntf = ext === ".resi";
        let mlname = isIntf ? name + ".mli" : name + ".ml";
```